adapters/Nend/NendAdapter/nativeads/GADMediationAdapterNendNativeForwarder.m

```diff
@@ -0,0 +1,53 @@
+//
+//  GADMediationAdapterNativeForwarder.m
+//  NendAdapter
+//
+//  Copyright © 2019 FAN Communications. All rights reserved.
+//
+
+#import "GADMediationAdapterNendNativeForwarder.h"
+#import "GADMAdapterNendConstants.h"
+#import "GADMAdapterNend.h"
+#import "GADMAdapterNendAdUnitMapper.h"
+
+@interface GADMediationAdapterNendNativeForwarder () <GADMAdapterNendNativeAdLoaderDelegate>
+
+@property(nonatomic, weak) id<GADMAdNetworkAdapter> adapter;
+@property(nonatomic, weak) id<GADMAdNetworkConnector> connector;
+
+@end
+
+@implementation GADMediationAdapterNendNativeForwarder
+
+
+- (instancetype)initWithAdapter:(id<GADMAdNetworkAdapter>)adapter
+                      connector:(id<GADMAdNetworkConnector>)connector {
+    self = [super init];
+    if (self != nil) {
+        _adapter = adapter;
+        _connector = connector;
+    }
+    return self;
+}
+
+- (void)getNativeAdWithAdTypes:(NSArray<GADAdLoaderAdType> *)adTypes options:(NSArray<GADAdLoaderOptions *> *)options {
+    id<GADMAdNetworkConnector> strongConnector = self.connector;
+
+    NSString *spotId= [GADMAdapterNendAdUnitMapper mappingAdUnitId:strongConnector paramKey:kGADMAdapterNendSpotID];
+    NSString *apiKey = [GADMAdapterNendAdUnitMapper mappingAdUnitId:strongConnector paramKey:kGADMAdapterNendApiKey];
+    GADMAdapterNendExtras *extras = [strongConnector networkExtras];
+
+    [self fetchNativeAd:options spotId:spotId apiKey:apiKey extra:extras];
+}
+
+- (void)didFailToLoadWithError:(NSError *)error {
+    id<GADMAdNetworkAdapter> strongAdapter = self.adapter;
+    [self.connector adapter:strongAdapter didFailAd:error];
+}
+
+- (void)didReceiveUnifiedNativeAd:(nonnull id<GADMediationNativeAd>)ad {
+    id<GADMAdNetworkAdapter> strongAdapter = self.adapter;
+    [self.connector adapter:strongAdapter didReceiveMediatedUnifiedNativeAd:ad];
+}
```

Comment:
Null check `strongAdapter`.
We should also create a `strongConnector` here as well and null check that.

Reply:
I'll check the object before attaching it.

Reply:
Should also create a `strongConnector`

```
id<GADMAdNetworkAdapter> strongAdapter = self.adapter;
id<GADMAdNetworkConnector> strongConnector = self.connector;

if (strongAdapter && strongConnector) {
  [strongConnector adapter:strongAdapter didFailAd:error];
}
```